userguide/tutorials/what_is_kill_bill.adoc

```diff
@@ -65,52 +65,55 @@ To know more, follow our https://docs.killbill.io/latest/userguide_deployment.ht
 
-
-As an example, let's imagine a Kill Bill deployment with two plugins, the first one a tax plugin that will interact with a 3rd party tax provider, and the second one, a payment plugin that will interact with a payment gateway.
+As an example, let's imagine a Kill Bill deployment with two plugins, the first one a tax plugin that will interact with a 3rd party tax provider, and the second one, a payment plugin, that will interact with a payment gateway.
 
 {empty} +
 
 [caption=""]
-//.Kill Bill: Big Picture
+//.Kill Bill: Deployment With Two Plugins
 image::../assets/what_is_killbill/KillBill_Plugins.png[format=png,align=center,title-align=center]
 
+These plugins are bundled as JAR files et deployed in the same Tomcat environment as Kill Bill. They can access the full spectrum of Kill Bill APIs, expose custom HTTP endpoints, or even access the database if they need to persist custom state.
+
+Any customization of Kill Bill will often require such plugins to be written. As such, it is not uncommon for deployments with custom business processes to have a custom of these, in addition to _connector_ plugins, which integrate with third-party systems.
+
 === Kill Bill Model
 
 Most billing systems out there have been designed to be batch oriented, i.e invoice generation and payments are made in batches at specific points in time. We have taken a different approach, and have built a system that is **event driven**, and where operations will happen in sequence.
 
-A typical scenario is one where a _subscription_ gets created, eventually leading to the creation of an _invoice_ and a _payment_. Reusing our example from above, i.e a deployment with the 2 plugins, the following would occur:
+A typical scenario is one where a _subscription_ gets created (e.g. a customer signs-up for a product), which leads to the creation of an _invoice_, which in turns triggers _payment_. Reusing our example from above, i.e. a deployment with the two plugins, the following would occur:
 
 {empty} +
 
 [caption=""]
-//.Kill Bill: Big Picture
+//.Kill Bill: Subscription Flow
 image::../assets/what_is_killbill/KillBill_Subscription_Flow.png[format=png,align=center,title-align=center]
 
-
 The flow diagram above highlights a couple of important aspects of the system:
 
-1. The creation of the subscription will generate a series of internal events that will lead to the generation of the _invoice_ and its associated _payment_. If needed such behavior can also be configured.
+1. The creation of the subscription will generate a series of internal events that will lead to the generation of the _invoice_ and its associated _payment_. If needed this default behavior can also be adjusted (e.g. to generate invoices in batches).
 2. Because of its plugin architecture, it is easy to extend the system by writing plugins. The diagrams above assumes the plugins interact with 3rd party systems, but of course all the tax logic could for instance be implemented in the plugin itself.
```

Comment:
```suggestion
2. Because of its plugin architecture, it is easy to extend the system by writing plugins. The diagrams above assumes the plugins interact with third-party systems, but of course all the tax logic could, for instance, be implemented in the plugin itself.
```